### PR TITLE
Jeanette/scoot135 log task data

### PR DESCRIPTION
**Pull request by JeanetteBruno**
updated taskRunner, etc to log the serialized processResults in the sagalog end task message

### CODE REVIEW

File: sched/scheduler/stateful_scheduler_test.go [hunk 1]
```diff
@@ -103,7 +103,9 @@ func Test_StatefulScheduler_JobRunsToCompletion(t *testing.T) {
 	sagaLogMock := saga.NewMockSagaLog(mockCtrl)
 	sagaLogMock.EXPECT().StartSaga(job.Id, nil)
 	sagaLogMock.EXPECT().LogMessage(saga.MakeStartTaskMessage(job.Id, taskId, nil))
-	sagaLogMock.EXPECT().LogMessage(saga.MakeEndTaskMessage(job.Id, taskId, nil))
+	endMessageMatcher := TaskMessageMatcher{JobId: job.Id, TaskId: taskId, Data: gomock.Any()}
+	sagaLogMock.EXPECT().LogMessage(endMessageMatcher)
+	//sagaLogMock.EXPECT().LogMessage(saga.MakeEndTaskMessage(job.Id, taskId, nil))
```
Comment (CaitieM20) [inline sched/scheduler/stateful_scheduler_test.go]: nit: remove unneeded line

File: sched/scheduler/task_runner_test.go [hunk 2]
```diff
@@ -22,7 +22,9 @@ func Test_runTaskAndLog_Successful(t *testing.T) {
 	sagaLogMock := saga.NewMockSagaLog(mockCtrl)
 	sagaLogMock.EXPECT().StartSaga("job1", nil)
 	sagaLogMock.EXPECT().LogMessage(saga.MakeStartTaskMessage("job1", "task1", nil))
-	sagaLogMock.EXPECT().LogMessage(saga.MakeEndTaskMessage("job1", "task1", nil))
+	endMessageMatcher := TaskMessageMatcher{JobId: "job1", TaskId: "task1", Data: gomock.Any()}
+	sagaLogMock.EXPECT().LogMessage(endMessageMatcher)
+	//sagaLogMock.EXPECT().LogMessage(saga.MakeEndTaskMessage("job1", "task1", nil))
```
Comment (CaitieM20) [inline sched/scheduler/task_runner_test.go]: nit: remove unneeded commented out line

File: common/thrifthelpers/thrift_serializer.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
+package thrifthelpers
+
+import "github.com/apache/thrift/lib/go/thrift"
+
+type Serializer interface {
```
Comment (CaitieM20) [inline common/thrifthelpers/thrift_serializer.go]: so in order to have an object that implements this interface for Json & Binary, you'll need to declare a struct and have Serialize and Deserialize be methods on it.  

type JsonSerializer struct {}
func (j JsonSerializer) Serialize ...
func (j JsonSerializer) Serialize ...

It looks like you do something like this in job_def_serializer.  Is this the piece that needs to be refactored next you mentioned?

**Comment by JeanetteBruno** [comment]
I'll remove the commented out lines.

wrt to the serializer question: thrift generates the structs that can be
serialized/deserialized.  Our use of Serialize/Deserialize will use the
generated structure as follows:
to serialize:
asBytes, err := thrifthelpers.<Json/Binary>Serialize(<thriftStructure>)

to deserialize:
err := thrifthelpers.<Json/Binary>Deserialize(<thriftStructure>,
sourceBytes)


see example at line 139 in workerapi/api.go

File: common/thrifthelpers/thrift_serializer.go (hunk 3 above)
Comment (CaitieM20) [inline common/thrifthelpers/thrift_serializer.go]: cool looks like interface isn't necessary then? So either remove the interface or add the structs up to you!

**Pull request merged**